src/securityPolicy.c

```diff
@@ -476,24 +484,27 @@ security_policy_t policyForSignTxCertificateStaking(
 		ASSERT(false);
 	}
 
-	if (stakingKeyPath) {
-		DENY_UNLESS(bip44_isOrdinaryStakingKeyPath(stakingKeyPath));
+	if (stakeCredential->type == STAKE_CREDENTIAL_KEY_PATH) {
+		DENY_UNLESS(bip44_isOrdinaryStakingKeyPath(&stakeCredential->pathSpec));
 	}
 
 	PROMPT();
 }
 
 security_policy_t policyForSignTxCertificateStakePoolRetirement(
         sign_tx_usecase_t signTxUsecase,
-        const bip44_path_t* poolIdPath,
+        const stake_credential_t* stakeCredential,
         uint64_t epoch MARK_UNUSED
 )
+// not sure about this one to be fair, the fact that we store
+// the pool id path in the stake credentials is a memory saving measure,
+// the policy wouldn't have to know about it
```

Comment:
Yeah, the policy should know nothing about that, it should only be given a path.

Reply:
Reworked.

Reply:
The comment should be removed, too.